src/Uno.UI.RuntimeTests/Tests/Windows_Storage/Given_StorageFile.cs

```diff
@@ -2,65 +2,173 @@
-
-        [TestCleanup]
-        public void Cleanup()
-        {
-        }
-
-        [TestMethod]
-        public async void When_DeleteFile()
-        {
-            
-            var _folder = Windows.Storage.ApplicationData.Current.LocalFolder;
-            Assert.IsNotNull(_folder, "cannot get LocalFolder - error outside tested method");
-
-            Windows.Storage.StorageFile _file = null;
-            
-            try
-            {
-                _file = await _folder.CreateFileAsync( _filename, Windows.Storage.CreationCollisionOption.FailIfExists);
-                Assert.IsNotNull(_file, "cannot create file - error outside tested method");
-            }
-            catch
-            {
-                  Assert.Fail("CreateFile exception - error outside tested method");
-            }
-
-            // try delete file
-            try
-            {
-                  await _file.DeleteAsync();
-            }
-            catch
-            {
-                Assert.Fail("DeleteAsync exception - error in tested method");
-            }
-
-            // check if method works
-            var _fileAfter = await _folder.TryGetItemAsync("test-deletingfile.txt");
-            Assert.IsNull(_fileAfter, "file is not deleted - tested method fails");
-      
-        }
-
-
-    }
+	[TestClass]
+	public class Given_StorageFile
+	{
+		String _filename;
+		
+		[TestInitialize]
+		public void Init()
+		{
+			_filename = "deletefile-" + DateTime.Now.ToString("yyyyMMddHHmmss") + ".txt";
+		}
+
+		[TestCleanup]
+		public void Cleanup()
+		{
+		}
+
+		[TestMethod]
+		public async Task When_DeleteFile()
+		{
+			
+			var _folder = Windows.Storage.ApplicationData.Current.LocalFolder;
+			Assert.IsNotNull(_folder, "cannot get LocalFolder - error outside tested method");
+
+			Windows.Storage.StorageFile _file = null;
+			
+			try
+			{
+				_file = await _folder.CreateFileAsync( _filename, Windows.Storage.CreationCollisionOption.FailIfExists);
+				Assert.IsNotNull(_file, "cannot create file - error outside tested method");
+			}
+			catch
+			{
+				  Assert.Fail("CreateFile exception - error outside tested method");
+			}
+
+			// try delete file
+			try
+			{
+				  await _file.DeleteAsync();
+			}
+			catch
+			{
+				Assert.Fail("DeleteAsync exception - error in tested method");
+			}
+
+			// check if method works
+			var _fileAfter = await _folder.TryGetItemAsync("test-deletingfile.txt");
+			Assert.IsNull(_fileAfter, "file is not deleted - tested method fails");
+		}
+
+		[TestMethod]
+		public async Task When_OpenRead()
+		{
+			var path = GetRandomFilePath();
+			var file = await (await GetFile(path)).OpenReadAsync();
+			var direct = File.Open(path, FileMode.OpenOrCreate, FileAccess.ReadWrite, FileShare.ReadWrite);
+
+			await StreamTestHelper.Test(writeTo: direct, readOn: file);
+
+			try
+			{
+				await StreamTestHelper.WriteData(file);
+				Assert.Fail("should have failed");
+			}
+			catch (NotSupportedException)
+			{
```

Comment:
Codacy found an issue: [Either remove or fill this block of code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6307767)